orm/sequence.go

```diff
@@ -6,50 +6,42 @@ import (
 	"github.com/iov-one/weave"
 )
 
-var seqPrefix = []byte("_s.")
-
 // Sequence maintains a counter, and generates a
 // series of keys. Each key is greater than the last,
 // both NextInt() as well as bytes.Compare() on NextVal().
 type Sequence struct {
 	id []byte
 }
 
-// NewSequence creates a sequence with this id
-// Form _s.<bucket>:<name>
-// KeyTagger uses _s.<bucket> as key
+// NewSequence returns a sequence counter. Sequence is using following pattern
+// to construct a key:
+//    _s.<bucket>:<name>
 func NewSequence(bucket, name string) Sequence {
-	suffix := bucket + ":" + name
+	id := "_s." + bucket + ":" + name
 	return Sequence{
-		id: append(seqPrefix, suffix...),
+		id: []byte(id),
 	}
 }
 
-// NextVal increments the sequence and returns next val as 8 bytes
+// NextVal increments the sequence and returns its state as 8 bytes.
 func (s *Sequence) NextVal(db weave.KVStore) []byte {
-	_, bz := s.increment(db)
+	_, bz := s.increment(db, 1)
 	return bz
 }
 
-// NextInt increments the sequence and returns next val as int
+// NextInt increments the sequence and returns its state as int.
 func (s *Sequence) NextInt(db weave.KVStore) int64 {
-	val, _ := s.increment(db)
+	val, _ := s.increment(db, 1)
 	return val
 }
 
-func (s *Sequence) curVal(db weave.KVStore) (key, val []byte) {
-	key = append(seqPrefix, s.id...)
-	val = db.Get(key)
-	return key, val
-}
-
-func (s *Sequence) increment(db weave.KVStore) (int64, []byte) {
-	key, bz := s.curVal(db)
-	val := decodeSequence(bz)
-	val++
-	bz = encodeSequence(val)
-	db.Set(key, bz)
-	return val, bz
+func (s *Sequence) increment(db weave.KVStore, inc int64) (int64, []byte) {
```

Comment:
I'm not sure what is the point of having inc parameter there?
If it's not used with values other than 1 - I'd remove this. Increment
without a value is known to increment by 1, so that does not need to be explicit

Reply:
Ok, I see now why you did this.
Disregard